sdk/helper/testcluster/docker/replication.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package docker
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/helper/logging"
+	"github.com/hashicorp/vault/sdk/helper/testcluster"
+)
+
+func NewReplicationSetDocker(t *testing.T) (*testcluster.ReplicationSet, error) {
+	binary := os.Getenv("VAULT_BINARY")
+	if binary == "" {
+		t.Skip("only running docker test when $VAULT_BINARY present")
+	}
+
+	r := &testcluster.ReplicationSet{
+		Clusters: map[string]testcluster.VaultCluster{},
+		Logger:   logging.NewVaultLogger(hclog.Trace).Named(t.Name()),
+	}
+
+	r.Builder = func(ctx context.Context, name string, baseLogger hclog.Logger) (testcluster.VaultCluster, error) {
+		cluster := NewTestDockerCluster(t, &DockerClusterOptions{
+			ImageRepo:   "hashicorp/vault",
+			ImageTag:    "latest",
+			VaultBinary: os.Getenv("VAULT_BINARY"),
+			ClusterOptions: testcluster.ClusterOptions{
+				ClusterName: strings.ReplaceAll(t.Name()+"-"+name, "/", "-"),
+				Logger:      baseLogger.Named(name),
+				VaultNodeConfig: &testcluster.VaultNodeConfig{
+					LogLevel: "TRACE",
+					// If you want the test to run faster locally, you could
+					// uncomment this performance_multiplier change.
+					//StorageOptions: map[string]string{
+					//	"performance_multiplier": "1",
+					//},
+				},
+			},
+			CA: r.CA,
+		})
+		t.Logf("Cluster %s started: VAULT_TOKEN=%s VAULT_CACERT=/vault/config/ca.pem", name, cluster.RootToken())
```

Comment:
This is very helpful!

Reply:
Yeah, but we need to also add something like this for after we generate a new token as part of enabling replication on secondary I think.  Might've seen something like that in your PR?

Reply:
Good point! I have not done that. I will add something. Though I have not passed in `t` to those functions. I will think of something.

Reply:
I decided to switch to using the cluster logger instead of t, see latest commit.